### PR TITLE
tests: support mock upstream driver

### PR DESCRIPTION
**Pull request by anandaravuri**

### CODE REVIEW

File: testing/fpgainfo/test_fpgainfo_c.cpp [hunk 1]
```diff
@@ -855,32 +846,33 @@ TEST_P(fpgainfo_c_p, bmc_filter0) {
...
-    free(tokens);
-    fpgaDestroyProperties(&filter);
+  char zero[20];
+  char one[20];
+  char *argv[] = {zero, one};
+
+  fpga_properties filter = NULL;
+  fpga_token *tokens = NULL;
+  uint32_t matches = 0, num_tokens = 0;
+  ;
```
Comment (jelonanX) [inline testing/fpgainfo/test_fpgainfo_c.cpp]: clang?

File: testing/fpgainfo/test_fpgainfo_c.cpp [hunk 2]
```diff
@@ -771,32 +763,33 @@ TEST_P(fpgainfo_c_p, temp_filter0) {
...
```
Comment (jelonanX) [inline testing/fpgainfo/test_fpgainfo_c.cpp]: clang?

File: testing/fpgainfo/test_fpgainfo_c.cpp [hunk 3]
```diff
@@ -687,32 +680,33 @@ TEST_P(fpgainfo_c_p, power_filter0) {
...
```
Comment (jelonanX) [inline testing/fpgainfo/test_fpgainfo_c.cpp]: clang?

File: testing/fpgainfo/test_fpgainfo_c.cpp [hunk 4]
```diff
@@ -519,32 +514,33 @@ TEST_P(fpgainfo_c_p, fme_filter0) {
...
```
Comment (jelonanX) [inline testing/fpgainfo/test_fpgainfo_c.cpp]: Clang?

File: testing/fpgainfo/test_fpgainfo_c.cpp [hunk 5]
```diff
@@ -401,85 +397,86 @@ TEST_P(fpgainfo_c_p, errors_filter2) {
...
+  char one[20];
+  char two[20];
+  char three[20];
+  char *argv3[] = {zero, one, two};
+  char *argv4[] = {zero, one, two, three};
+
+  fpga_properties filter = NULL;
+  fpga_token *tokens = NULL;
+  uint32_t matches = 0, num_tokens = 0;
+  ;
```
Comment (jelonanX) [inline testing/fpgainfo/test_fpgainfo_c.cpp]: This might be from clang, can we remove this?

File: libopae/plugins/xfpga/enum.c [hunk 6]
```diff
@@ -386,7 +386,7 @@ STATIC fpga_result enum_fme(const char *sysfspath, const char *name,
 	// Discover the socket id from the FME's sysfs entry.
 	resval = sysfs_parse_attribute64(sysfspath, FPGA_SYSFS_SOCKET_ID, &value);
 	if (resval != 0) {
-		return FPGA_NOT_FOUND;
+		//return FPGA_NOT_FOUND;
```
Comment (tswhison) [inline libopae/plugins/xfpga/enum.c]: Why is this being commented out?

**Pull request closed without merging**